filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilder.scala

```diff
@@ -55,39 +54,39 @@ object BlobPathBuilder {
   }
 }
 
-class BlobPathBuilder(blobTokenGenerator: BlobTokenGenerator, container: String, endpoint: String) extends PathBuilder {
-
-  val credential: AzureSasCredential = new AzureSasCredential(blobTokenGenerator.getAccessToken)
-  val fileSystemConfig: Map[String, Object] = Map((AzureFileSystem.AZURE_STORAGE_SAS_TOKEN_CREDENTIAL, credential),
-                                                  (AzureFileSystem.AZURE_STORAGE_FILE_STORES, container),
-                                                  (AzureFileSystem.AZURE_STORAGE_SKIP_INITIAL_CONTAINER_CHECK, java.lang.Boolean.TRUE))
-
-  def retrieveFilesystem(uri: URI): Try[FileSystem] = {
-    Try(FileSystems.getFileSystem(uri)) recover {
-      // If no filesystem already exists, this will create a new connection, with the provided configs
-      case _: FileSystemNotFoundException => FileSystems.newFileSystem(uri, fileSystemConfig.asJava)
-    }
-  }
+class BlobPathBuilder(fsm: FileSystemManager, container: String, endpoint: String) extends PathBuilder {
 
   def build(string: String): Try[BlobPath] = {
     validateBlobPath(string, container, endpoint) match {
-      case ValidBlobPath(path) => for {
-            fileSystem <- retrieveFilesystem(new URI("azb://?endpoint=" + endpoint))
-            nioPath <- Try(fileSystem.getPath(path))
-            blobPath = BlobPath(nioPath, endpoint, container)
-          } yield blobPath
+      case ValidBlobPath(path) => Try(BlobPath(path, endpoint, container, fsm))
       case UnparsableBlobPath(errorMessage: Throwable) => Failure(errorMessage)
     }
   }
-
   override def name: String = "Azure Blob Storage"
 }
 
-// Add args for container, storage account name
-case class BlobPath private[blob](nioPath: NioPath, endpoint: String, container: String) extends Path {
-  override protected def newPath(nioPath: NioPath): Path = BlobPath(nioPath, endpoint, container)
+case class BlobPath private[blob](pathString: String, endpoint: String, container: String, fsm: FileSystemManager) extends Path {
```

Comment:
It's an interesting idea to put it in a separate set of args, not sure it gains much by being implicit. I think this is a little-used but valid pattern:
```
case class BlobPath(pathString: String, endpoint: String, container: String)(private val fsm: FileSystemManager)
```
This might be something that's a little easier to think about once the tests are in place.